scrapy_zyte_api/_request_fingerprinter.py

```diff
@@ -36,30 +37,28 @@ def __init__(self, crawler):
             )
             self._cache: "WeakKeyDictionary[Request, bytes]" = WeakKeyDictionary()
             self._param_parser = _ParamParser(crawler, cookies_enabled=False)
-            self._skip_keys = (
-                "customHttpRequestHeaders",
-                "echoData",
-                "jobId",
-                "requestHeaders",
-                "experimental",
-            )
 
-        def _keep_fragments(self, api_params):
-            return any(
-                api_params.get(key, False) for key in ("browserHtml", "screenshot")
+        def _normalize_params(self, api_params):
+            api_params["url"] = canonicalize_url(
+                api_params["url"],
+                keep_fragments=_uses_browser(api_params),
             )
 
+            if "httpRequestText" in api_params:
+                api_params["httpRequestBody"] = b64encode(
+                    api_params.pop("httpRequestText").encode()
+                ).decode()
```

Comment:
should we pop httpRequestText in this case?

Reply:
Note that we are only popping for the purpose of fingerprinting, `api_params` is a copy of the original params.

Reply:
Yes, the idea is that the request with httpRequestText and the same httpRequestBody have the same fingerprint